test/test_irods_http_api.py

```diff
@@ -709,6 +709,122 @@ def test_common_operations(self):
+        self.assertEqual(r.json()['irods_response']['status_code'], 0)
+
+        # Create a non-empty data object.
+        data_object = os.path.join('/', self.zone_name, 'home', self.rodsuser_username, 'common_ops.txt')
+        content = 'hello, this message was written via the iRODS HTTP API!'
+        r = requests.post(self.url_endpoint, headers=rodsuser_headers, data={
+            'op': 'write',
+            'lpath': data_object,
+            'bytes': content,
+            'count': len(content)
+        })
+        #print(r.content) # Debug
+        self.assertEqual(r.status_code, 200)
+        self.assertEqual(r.json()['irods_response']['status_code'], 0)
+
+        # Replicate the data object.
+        r = requests.post(self.url_endpoint, headers=rodsuser_headers, data={
+            'op': 'replicate',
+            'lpath': data_object,
+            'dst-resource': resc_name
+        })
+        #print(r.content) # Debug
+        self.assertEqual(r.status_code, 200)
+        self.assertEqual(r.json()['irods_response']['status_code'], 0)
+
+        # Show there are two replicas.
+        coll_name = os.path.dirname(data_object)
+        data_name = os.path.basename(data_object)
+        r = requests.get(f'{self.url_base}/query', headers=rodsuser_headers, params={
+            'op': 'execute_genquery',
+            'query': f"select DATA_NAME, RESC_NAME where COLL_NAME = '{coll_name}' and DATA_NAME = '{data_name}'"
+        })
+        #print(r.content) # Debug
+        self.assertEqual(r.status_code, 200)
+        result = r.json()
+        self.assertEqual(result['irods_response']['status_code'], 0)
+        self.assertEqual(len(result['rows']), 2)
+
+        # Calculate a checksum for first replica.
+        r = requests.post(self.url_endpoint, headers=rodsuser_headers, data={
+            'op': 'calculate_checksum',
+            'lpath': data_object,
+            'replica-number': 0
+        })
+        #print(r.content) # Debug
+        self.assertEqual(r.status_code, 200)
+        result = r.json()
+        self.assertEqual(result['irods_response']['status_code'], 0)
+        self.assertGreater(len(result['checksum']), 0)
+        self.assertEqual(result['checksum'][:5], 'sha2:')
+
+        # Verify checksum information across all replicas.
+        r = requests.get(self.url_endpoint, headers=rodsuser_headers, params={
+            'op': 'verify_checksum',
+            'lpath': data_object
+        })
+        #print(r.content) # Debug
+        self.assertEqual(r.status_code, 200)
+        result = r.json()
+        self.assertEqual(result['irods_response']['status_code'], -407000) # CHECK_VERIFICATION_RESULTS
+        self.assertEqual(result['results'][0]['error_code'], -862000) # CAT_NO_CHECKSUM_FOR_REPLICA
+        self.assertEqual(result['results'][0]['message'], 'WARNING: No checksum available for replica [1].')
+        self.assertEqual(result['results'][0]['severity'], 'warning')
+
```

Comment:
Yes, we can, but I'm not sure it adds anything.

The test proves we're hitting the correct endpoint in iRODS and receiving the correct information. I think that is good enough for now, otherwise, we'd be re-implementing test cases that are already covered by the iRODS server.

Perhaps calculating a checksum for the second replica is better. That at least proves the HTTP API supports selecting specific replicas.

Reply:
that's fair.   i was thinking about making sure we get a 'good' response when they're both calculated/present... so we don't get the error case coming back.

Reply:
Cool cool. Resolving now.